base/reshapedarray.jl

```diff
@@ -107,16 +107,15 @@ julia> reshape(1:6, 2, 3)
 """
 reshape
 
-reshape(parent::AbstractArray, dims::IntOrInd...) = reshape(parent, dims)
+reshape(parent::AbstractArray, dims::Union{Integer, Colon, AbstractUnitRange{<:Integer}}...) = reshape(parent, dims)
+reshape(parent::AbstractArray, shp::Tuple{Integer, Vararg{Integer}}) = reshape(parent, to_shape(shp))
 reshape(parent::AbstractArray, shp::Tuple{Union{Integer,OneTo}, Vararg{Union{Integer,OneTo}}}) = reshape(parent, to_shape(shp))
 reshape(parent::AbstractArray, dims::Dims)        = _reshape(parent, dims)
 
 # Allow missing dimensions with Colon():
 reshape(parent::AbstractVector, ::Colon) = parent
 reshape(parent::AbstractVector, ::Tuple{Colon}) = parent
-reshape(parent::AbstractArray, dims::Int...) = reshape(parent, dims)
-reshape(parent::AbstractArray, dims::Union{Int,Colon}...) = reshape(parent, dims)
-reshape(parent::AbstractArray, dims::Tuple{Vararg{Union{Int,Colon}}}) = reshape(parent, _reshape_uncolon(parent, dims))
+reshape(parent::AbstractArray, dims::Tuple{Vararg{Union{Integer,Colon,Base.OneTo}}}) = reshape(parent, _reshape_uncolon(parent, dims))
```

Comment:
This feels like it should be:
```suggestion
reshape(parent::AbstractArray, dims::Tuple) = reshape(parent, _reshape_uncolon(parent, dims))
```
wdyt?

Reply:
I like it, although perhaps worth mentioning that this almost makes `_reshape_uncolon` part of the API (it becomes a tempting target for package specialization).